src/main/kotlin/org/jetbrains/research/ml/codetracker/ui/panes/SurveyPane.kt

```diff
@@ -196,11 +204,25 @@ class SurveyController(project: Project, scale: Double, fxPanel: JFXPanel, id: I
 
     private fun initCountry() {
 //        Todo: make it autocomplete https://stackoverflow.com/questions/19924852/autocomplete-combobox-in-javafx
-        countryObservableList = FXCollections.observableList(paneUiData.country.dataList.map {
-            it.translation[paneUiData.language.currentValue]
-        })
+        countryObservableList = FXCollections.observableList(paneUiData.country.dataList)
         countryComboBox.items = countryObservableList
 
+        val cellFactory = Callback<ListView<Country>, ListCell<Country>> {
+            object : ListCell<Country>() {
+                override fun updateItem(item: Country?, empty: Boolean) {
+                    super.updateItem(item, empty)
+                    if (item == null || empty) {
+                        graphic = null;
+                    } else {
+                        text = item.translation.getOrDefault(LanguagePaneUiData.language.currentValue, "")
+                    }
+                }
+            }
+        }
+
+        countryComboBox.setButtonCell(cellFactory.call(null))
+        countryComboBox.setCellFactory(cellFactory)
```

Comment:
Could you use properties, please? For example, 
`countryComboBox.buttonCell = cellFactory.call(null)`

It looks prettier, I think

Reply:
You're absolutely right, I just copypasted it from StackOverflow and forgot to change lol :)